### PR TITLE
feat(eg/channel): add new data source to query official channels

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Add a new data source to query official channels.
image

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new data source to query official channels.
```

## PR Checklist

* [x] Tests added/passed.

```
../coverage.sh -o eg -f TestAccDataOfficialEventChannels_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/eg" -v -coverprofile="./huaweicloud/services/acceptance/eg_coverage.cov" -coverpkg="./huaweicloud/services/eg" -run TestAccDataOfficialEventChannels_basic -timeout 360m -parallel 10
=== RUN   TestAccDataOfficialEventChannels_basic
--- PASS: TestAccDataOfficialEventChannels_basic (35.69s)
PASS
coverage: 8.4% of statements in ./huaweicloud/services/eg
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        35.749s coverage: 8.4% of statements in ./huaweicloud/services/eg
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The region where the official event channels are located.`,
+			},
+			"channel_id": {
```
Comment (deer-hang) [inline huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go]: It seems that there is no need to add this field.
It is recommended to remove it.

File: docs/data-sources/eg_official_event_channels.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region where the official event channels are located.  
+  If omitted, the provider-level region will be used.
+
+* `name` - (Optional, String) Specifies the channel name used to query specified official event channel.
+
```
Comment (deer-hang) [inline docs/data-sources/eg_official_event_channels.md]: Why fields `provider_type`, `sort` and `fuzzy_name` are not supported?

File: docs/data-sources/eg_official_event_channels.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+<a name="eg_official_event_channels"></a>
+The `channels` block supports:
+
+* `id` - The ID of the official event channel.
+
+* `name` - The name of the official event channel.
+
+* `description` - The description of the official event channel.
+
+* `provider_type` - The type of the official event channel.
```
Comment (deer-hang) [inline docs/data-sources/eg_official_event_channels.md]: Please list the enumeration values ​​for this field.

File: huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The type of the official event channel.`,
+						},
+						"enterprise_project_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The ID of the enterprise project to which the official event channel belongs.`,
+						},
+						"cross_account_ids": {
```
Comment (deer-hang) [inline huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go]: It looks like this field should be removed.

File: huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(uuid)
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("channels", flattenDataEventChannels(filterDataEventChannels(cfg, d, channels))),
```
Comment (deer-hang) [inline huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go]: Method `flattenDataEventChannels` lacks a method to backfill **description**.

File: huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go]: sure

File: docs/data-sources/eg_official_event_channels.md (hunk 2 above)
Comment (Lance52259) [reply docs/data-sources/eg_official_event_channels.md]: filter parameter `provider_type` can be supported.

File: docs/data-sources/eg_official_event_channels.md (hunk 3 above)
Comment (Lance52259) [reply docs/data-sources/eg_official_event_channels.md]: all enums are explained in the description of the corresponding filter parameter

File: huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go]: this attributes is take effect for the custom event channels.

File: huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go]: yeah, description has been supported now

**Comment by Lance52259** [comment]
```
../coverage.sh -o eg -f TestAccDataEventChannels_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/eg" -v -coverprofile="./huaweicloud/services/acceptance/eg_coverage.cov" -coverpkg="./huaweicloud/services/eg" -run TestAccDataEventChannels_basic -timeout 360m -parallel 10
=== RUN   TestAccDataEventChannels_basic
--- PASS: TestAccDataEventChannels_basic (54.81s)
PASS
coverage: 15.8% of statements in ./huaweicloud/services/eg
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        54.928s coverage: 15.8% of statements in ./huaweicloud/services/eg
```

**Pull request merged**